src/main/java/de/tum/in/www1/artemis/domain/ProgrammingExercise.java

```diff
@@ -88,6 +89,10 @@ public class ProgrammingExercise extends Exercise {
     @Column(name = "project_key", table = "programming_exercise_details", nullable = false)
     private String projectKey;
 
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "build_plan_id", table = "exercise")
+    private BuildPlan buildPlan;
```

Comment:
we explicitly avoided this relationship, because this will load the build plan automatically for each programming exercise even for cases in which it will not be used, which will significantly decrease the performance in many cases.

Unfortunately, `ManyToOne` loads eager, even if you specify it as lazy. 

Therefore, we cannot merge this PR if this change is active, we have to find another solution.